internal/dcache/clustermap/clustermap.go

```diff
@@ -449,6 +459,46 @@ func (c *ClusterMap) getOfflineMVs() map[string]dcache.MirroredVolume {
 	return offlineMVs
 }
 
+func (c *ClusterMap) getActiveMVsForRV(rvName string) map[string]struct{} {
+	//
+	// Save a copy of the clusterMap pointer to use for accessing RVMap and MVMap, so that both
+	// correspond to the same instance of clusterMap.
+	//
+	localMap := c.getLocalMap()
+
+	rv, ok := localMap.RVMap[rvName]
+	if !ok {
+		common.Assert(false, rvName)
+		return nil
+	}
+
+	rvIsOffline := (rv.State == dcache.StateOffline)
+
+	activeMVs := make(map[string]struct{})
+	for mvName, mv := range localMap.MVMap {
+		// RV is offline in the RV list, so it cannot have any active MVs.
+		if rvIsOffline {
+			continue
+		}
+
+		rvState, ok := mv.RVs[rvName]
+		if !ok {
+			// RV is not a component of this MV (maybe it was in the past).
+			continue
+		}
+
+		// If the component RV is offline/inband-offline, then the MV is not active.
+		if rvState == dcache.StateOffline || rvState == dcache.StateInbandOffline {
+			continue
+		}
+
+		// Else, it's an active MV for this RV.
+		activeMVs[mvName] = struct{}{}
```

Comment:
Should we only add the MV to the map only when the RV's state in the MV is online. We should avoid outofsync and syncing states also here.